ivy/functional/frontends/tensorflow/general_functions.py

```diff
@@ -650,3 +651,21 @@ def truncatemod(x, y):
 @to_ivy_arrays_and_back
 def unravel_index(indices, dims, out=None, name=None):
     return ivy.unravel_index(indices, dims, out=out)
+
+@to_ivy_arrays_and_back
+def sequence_mask(lengths, maxlen=None, dtype=ivy.bool, name=None):
+
+    if maxlen is None:
+        maxlen = ivy.maximum(ivy.max(lengths), ivy.max(ivy.arange(ivy.get_num_dims(lengths))))
+        maxlen = ivy.maximum(0, maxlen)
+    else:
+        maxlen = ivy.array(maxlen)
+    if ivy.get_num_dims(maxlen )is not None and ivy.get_num_dims(maxlen) != 0:
+        raise ValueError("Argument `maxlen` must be scalar for sequence_mask, "
+                         f"received `maxlen` = {maxlen} "
+                         f"with shape '{maxlen.get_shape()}' instead")
+
+    row_vector = ivy.arange(0, int(maxlen), 1)
+    matrix = cast(ivy.expand_dims(lengths, axis=-1), maxlen.dtype)._ivy_array
```

Comment:
Are you sure that using `_ivy_array` is a good practice?

Reply:
The `cast` function outputs an eager tensor, and it's not possible to perform the less than operation on that in the next line. It throws an error. So I convert the `matrix` into an Ivy Array which makes it possible to find the result later.

Reply:
I guess it would be better if you used:
```python
ivy.astype(ivy.expand_dims(lengths, axis=-1), maxlen.dtype)
OR
ivy.expand_dims(lengths, axis=-1).astype(maxlen.dtype)
```
Does those code fragments have issues?

Reply:
I removed the cast function as a whole, as it was just a precautionary measure.